base/compiler/ssair/inlining.jl

```diff
@@ -1336,9 +1343,14 @@ function handle_any_const_result!(cases::Vector{InliningCase},
     allow_abstract::Bool, allow_typevars::Bool)
     if isa(result, ConcreteResult)
         return handle_concrete_result!(cases, result, state)
-    elseif isa(result, SemiConcreteResult)
-        return handle_semi_concrete_result!(cases, result; allow_abstract)
-    elseif isa(result, ConstPropResult)
+    end
+    if isa(result, SemiConcreteResult)
+        result = inlining_policy(state.interp, result, info, flag, result.mi, argtypes)
```

Comment:
So does `inlining_policy(::ExternalAbstractInterpreter, ::SemiConcreteResult, ...)` may return `ConstPropResult` instead of `SemiConcreteResult`?

Reply:
Mostly it'd return `nothing`, but I didn't see a good reason to forbid `ConstPropResult`.

Reply:
Well, it may also return `ConcreteResult` (e.g. in a case when the IR-interpretation figures out that the call is :foldable in a presence of non-const arguments), right? I wonder if we want to generalize `inlining_policy` so that it handles `SemiConcreteResult`/`ConcreteResult`/`ConstPropResult`.

Reply:
Hmm, I suppose, but I don't actually have a current use case where it does anything other than return `nothing` or pass through the SemiConcreteResult as is. I think I'd rather just only allow those two cases then until we have an actual use for it.